pkg/executor/fscache/poolcache.go

```diff
@@ -235,6 +240,17 @@ func (c *PoolCache) service() {
 		case deleteValue:
 			delete(c.cache[req.function].svcs, req.address)
 			req.responseChannel <- resp
+		case logFuncSvc:
+			var info io.Writer = req.file
+			defer req.file.Close()
+			for _, fnSvcGrp := range c.cache {
```

Comment:
You can use https://pkg.go.dev/bufio#NewWriter bufio.Writer to wrap file writer.
And keep writing data.
Once done writing flush buffer which will ensure data is written to the file.